test/controllers/formDataSubmit.test.js

```
@@ -0,0 +1,148 @@
+            const grade = JSONPath.value(response, "$..components[?(@.key=='grade')].defaultValue");
+            const personId = JSONPath.value(response, "$..components[?(@.key=='personid')].defaultValue");
+
+            expect(response.status).to.equal(400);
+            expect(grade).to.equal("{$.staffDetailsDataContext.gradeid}");
+            expect(personId).to.equal("{$.staffDetailsDataContext.staffid}");
+      });
+      it('should throw exception if 500 received from formio', async () => {
+            nock('http://localhost:8000', {})
+              .log(console.log)
+              .post('/form/formId/submission', {
+                data: {
+                  field1: "foo",
+                  field2: "bar"
+                }
+              })
+              .reply(500);
+            const request = httpMocks.createRequest({
+                method: 'POST',
+                url: '/api/translation/form/formId/submission',
+                body: {
+                  data: {
+                    field1: "foo",
+                    field2: "bar"
+                  }
+                },
+                params: {
+                  formId: "formId"
+                },
+                kauth: {
+                    grant: {
+                        access_token: {
+                            token: "test-token",
+                            content: {
+                                session_state: "session_id",
+                                email: "emailTest123",
+                                preferred_username: "test",
+                                given_name: "firstname",
+                                family_name: "surname"
+                            }
+                        }
+
+                    }
+                }
+            });
+            try {
+              await formTranslateController.submitForm(request);
+              return Promise.reject("Shouldn't get here");
```

We're not expecting a promise to be returned here so would probably be better to throw a standard error.

---

I tried that; the test passed when it shouldn't.

---

Ok fair enough...I guess this is because the test is an `async` function.